src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,85 @@
+import { FC, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { updateTodo } from '../../Api/todos';
+
+interface Props {
+  todo: Todo,
+  onDelete: (id: number) => void,
+  changeStatus: (id: number, property: Partial<Todo>) => void,
+}
+
+export const TodoItem: FC<Props> = ({ todo, onDelete, changeStatus }) => {
+  const {
+    title,
+    completed,
+    id,
+  } = todo;
+  const [todoTitle, setTitle] = useState(title);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const submitEditing = () => {
+    if (!todoTitle) {
+      onDelete(id);
+    } else {
+      updateTodo(id, { title: todoTitle });
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    submitEditing();
+    setIsEditing(false);
+  };
+
+  return (
+    <div className={classNames('todo', { completed })}>
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          checked={completed}
+          onClick={() => changeStatus(id,
+            { completed: !completed })}
+        />
+      </label>
+
+      { isEditing
+        ? (
+          <form onSubmit={handleSubmit}>
+            <input
+              className="todo__title todo__title-field"
+              type="text"
+              value={todoTitle}
+              onChange={(event) => setTitle(event.target.value)}
+              onBlur={submitEditing}
+            />
+          </form>
+        )
+        : (
+          <span
+            className="todo__title"
+            onDoubleClick={() => {
+              setIsEditing(true);
+            }}
+          >
+            {todoTitle}
+          </span>
+        )}
+
+      <button
+        type="button"
+        className="todo__remove"
+        onClick={() => onDelete(id)}
+      >
+        ×
+      </button>
+
+      {/* overlay will cover the todo while it is being updated */}
+      <div className="modal overlay">
```

Comment:
The modal overlay should only be visible while the API request is ongoing. You can use a state variable to control its visibility. 👁️